File: src/test/java/com/datarepublic/testexercise/ebayframework/BaseWidget.java
package com.datarepublic.testexercise.ebayframework;

import org.openqa.selenium.WebElement;

public class BaseWidget
{
    protected final WebElement boundingElem;
    
    public BaseWidget(WebElement elem)
    {
        if (elem == null)
            throw new IllegalArgumentException("elem cannot be null");
        boundingElem = elem;
    }
}

File: net/TestExerciseEbay/EbayPOMFramework/BaseWidget.cs
using System;
using OpenQA.Selenium;

namespace TestExercise.EbayPOMFramework
{
    public class BaseWidget
    {
        protected readonly IWebElement boundingElem;
    
    public BaseWidget(IWebElement elem)
        {
            if (elem == null)
                throw new ArgumentNullException("elem cannot be null");
            boundingElem = elem;
        }
    }
}